aplib.net-demo/Assets/Scripts/Models/Model.cs

```diff
@@ -0,0 +1,14 @@
+using Assets.Scripts.WFC;
+using System;
+
+namespace Assets.Scripts.Models
+{
+    public class Model
+    {
+        private readonly Func<int, int, Grid, bool> _func;
+
+        public Model(Func<int, int, Grid, bool> func) => _func = func;
```

Comment:
Perhaps add a private `GameObject` so that the algorithms knows what model to place. Perhaps also an orientation?

```suggestion
        private readonly Func<int, int, Grid, bool> _func;
		
		private readonly GameObject _model;
		
        public Model(GameObject model, Func<int, int, Grid, bool> func)
        {
        	_func = func;
			_model = model;
        }
```